### ISSUE TITLE
Improve security by connecting to APT repos over TLS (https)

### ISSUE THREAD

**Issue by kpapad904** [hestiacp/hestiacp#695]
Considering that:
"It is a common misconception that simply signing your packages and repository metadata with GPG is enough to create a secure APT repository. This is false. Many of the attacks outlined in the paper and this blog post are effective against GPG-signed APT repositories. GPG signing Debian packages themselves does nothing, as explained below. **The easiest way to prevent the attacks covered below is to always serve your APT repository over TLS; no exceptions**."  -- https://blog.packagecloud.io/eng/2018/02/21/attacks-against-secure-apt-repositories/

However, based on some quick testing, neither the MariaDB nor the PostgreSQL official repo mirrors can be accessed over https :-(

So for the moment the HestiaCP installer can be changed to retrieve over https the nginx signing key (important) and probably the repo:

```
diff install/hst-install-debian.sh*
602,603c602,603
< echo "deb [arch=amd64] http://nginx.org/packages/mainline/$VERSION/ $codename nginx" > $apt/nginx.list
< wget --quiet http://nginx.org/keys/nginx_signing.key -O /tmp/nginx_signing.key
---
> echo "deb [arch=amd64] https://nginx.org/packages/mainline/$VERSION/ $codename nginx" > $apt/nginx.list
> wget --quiet https://nginx.org/keys/nginx_signing.key -O /tmp/nginx_signing.key
```

**Comment by kpapad904** [comment]
TL;DR: The install/hst-install-debian.sh script should just use "apt-key adv --recv-keys --keyserver keyserver.ubuntu.com keyid" to retrieve the signing keys of all 5 (hestiacp, nginx, php, mariadb, postgresdb) APT repos used by HestiaCP.

The apt-key man page says: “Note that there are no checks performed, so it is easy to completely undermine the apt-secure(8) infrastructure if used without care.” This means **you really don't want to wget the nginx_signing.key without SSL**.

So retrieving the keys using "apt-key adv --fetch-keys https://…" is probably a good idea.
```
root@vm03:~# apt-key adv --fetch-keys 'https://mariadb.org/mariadb_release_signing_key.asc'
Executing: /tmp/apt-key-gpghome.1w68yjSsgP/gpg.1.sh --fetch-keys https://mariadb.org/mariadb_release_signing_key.asc
gpg: requesting key from 'https://mariadb.org/mariadb_release_signing_key.asc'
gpg: key F1656F24C74CD1D8: 6 signatures not checked due to missing keys
gpg: key F1656F24C74CD1D8: "MariaDB Signing Key <signing-key@mariadb.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
root@vm03:~# apt-key adv --fetch-keys 'https://nginx.org/keys/nginx_signing.key'
Executing: /tmp/apt-key-gpghome.VkMfoINgTm/gpg.1.sh --fetch-keys https://nginx.org/keys/nginx_signing.key
gpg: requesting key from 'https://nginx.org/keys/nginx_signing.key'
gpg: key ABF5BD827BD9BF62: 3 signatures not checked due to missing keys
gpg: key ABF5BD827BD9BF62: "nginx signing key <signing-key@nginx.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
root@vm03:~# apt-key adv --fetch-keys 'https://packages.sury.org/php/apt.gpg'
Executing: /tmp/apt-key-gpghome.kPYHPp7OXs/gpg.1.sh --fetch-keys https://packages.sury.org/php/apt.gpg
gpg: requesting key from 'https://packages.sury.org/php/apt.gpg'
gpg: key B188E2B695BD4743: "DEB.SURY.ORG Automatic Signing Key <deb@sury.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
root@vm03:~# apt-key adv --fetch-keys 'https://gpg.hestiacp.com/deb_signing.key'
Executing: /tmp/apt-key-gpghome.1ipt6dd6WQ/gpg.1.sh --fetch-keys https://gpg.hestiacp.com/deb_signing.key
gpg: requesting key from 'https://gpg.hestiacp.com/deb_signing.key'
gpg: key 70BCF820C9B5A99E: "HestiaCP <info@hestiacp.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
root@vm03:~# 
```

or retrieve all the keys from a keyserver:
```
root@vm03:~# apt-key adv --recv-keys --keyserver keyserver.ubuntu.com B188E2B695BD4743
Executing: /tmp/apt-key-gpghome.UgHDpQqpvc/gpg.1.sh --recv-keys --keyserver keyserver.ubuntu.com B188E2B695BD4743
gpg: key B188E2B695BD4743: 1 signature not checked due to a missing key
gpg: key B188E2B695BD4743: "DEB.SURY.ORG Automatic Signing Key <deb@sury.org>" 1 new signature
gpg: Total number processed: 1
gpg:         new signatures: 1
root@vm03:~# apt-key adv --recv-keys --keyserver keyserver.ubuntu.com ABF5BD827BD9BF62
Executing: /tmp/apt-key-gpghome.Lz8B8qOUGx/gpg.1.sh --recv-keys --keyserver keyserver.ubuntu.com ABF5BD827BD9BF62
gpg: key ABF5BD827BD9BF62: 13 signatures not checked due to missing keys
gpg: key ABF5BD827BD9BF62: "nginx signing key <signing-key@nginx.com>" 11 new signatures
gpg: Total number processed: 1
gpg:         new signatures: 11
root@vm03:~# apt-key adv --recv-keys --keyserver keyserver.ubuntu.com F1656F24C74CD1D8
Executing: /tmp/apt-key-gpghome.CfXHLWtDtM/gpg.1.sh --recv-keys --keyserver keyserver.ubuntu.com F1656F24C74CD1D8
gpg: key F1656F24C74CD1D8: 6 signatures not checked due to missing keys
gpg: key F1656F24C74CD1D8: "MariaDB Signing Key <signing-key@mariadb.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
root@vm03:~# 
```

either method is apparently better than using wget / apt-key add command pairs.

Finally, the APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 currently used by the HestiaCP installer, doesn't seem to be needed:
```
root@vm03:~# apt-key del ABF5BD827BD9BF62 F1656F24C74CD1D8 B188E2B695BD4743
OK
root@vm03:~# apt-key list
...
root@vm03:~# apt-key adv --recv-keys --keyserver keyserver.ubuntu.com B188E2B695BD4743
Executing: /tmp/apt-key-gpghome.cHem5y2wCm/gpg.1.sh --recv-keys --keyserver keyserver.ubuntu.com B188E2B695BD4743
gpg: key B188E2B695BD4743: 1 signature not checked due to a missing key
gpg: key B188E2B695BD4743: public key "DEB.SURY.ORG Automatic Signing Key <deb@sury.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
root@vm03:~# apt-key adv --recv-keys --keyserver keyserver.ubuntu.com ABF5BD827BD9BF62
Executing: /tmp/apt-key-gpghome.cmaIkI04ae/gpg.1.sh --recv-keys --keyserver keyserver.ubuntu.com ABF5BD827BD9BF62
gpg: key ABF5BD827BD9BF62: 13 signatures not checked due to missing keys
gpg: key ABF5BD827BD9BF62: public key "nginx signing key <signing-key@nginx.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
root@vm03:~# apt-key adv --recv-keys --keyserver keyserver.ubuntu.com F1656F24C74CD1D8
Executing: /tmp/apt-key-gpghome.fQZaYOIYpD/gpg.1.sh --recv-keys --keyserver keyserver.ubuntu.com F1656F24C74CD1D8
gpg: key F1656F24C74CD1D8: 6 signatures not checked due to missing keys
gpg: key F1656F24C74CD1D8: public key "MariaDB Signing Key <signing-key@mariadb.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
root@vm03:~# 
```

Hope this helps and please keep up the great work!

**Comment by kpapad904** [comment]
Here's the installer using the apt-key adv --fetch-keys method (note: HestiaCP's own signing key 70BCF820C9B5A99E isn't available from the keyservers):

```
$ diff install/hst-install-debian.sh*
602,604c602,603
< echo "deb [arch=amd64] http://nginx.org/packages/mainline/$VERSION/ $codename nginx" > $apt/nginx.list
< wget --quiet http://nginx.org/keys/nginx_signing.key -O /tmp/nginx_signing.key
< APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key add /tmp/nginx_signing.key > /dev/null 2>&1
---
> echo "deb [arch=amd64] https://nginx.org/packages/mainline/$VERSION/ $codename nginx" > $apt/nginx.list
> apt-key adv --fetch-keys 'https://nginx.org/keys/nginx_signing.key' > /dev/null 2>&1
609,610c608
< wget --quiet https://packages.sury.org/php/apt.gpg -O /tmp/php_signing.key
< APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key add /tmp/php_signing.key > /dev/null 2>&1
---
> apt-key adv --fetch-keys 'https://packages.sury.org/php/apt.gpg' > /dev/null 2>&1
616c614
<     APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key adv --recv-keys --keyserver keyserver.ubuntu.com CBCB082A1BB943DB > /dev/null 2>&1
---
>     apt-key adv --recv-keys --keyserver keyserver.ubuntu.com CBCB082A1BB943DB > /dev/null 2>&1
618c616
<     APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key adv --recv-keys --keyserver keyserver.ubuntu.com F1656F24C74CD1D8 > /dev/null 2>&1
---
>     apt-key adv --recv-keys --keyserver keyserver.ubuntu.com F1656F24C74CD1D8 > /dev/null 2>&1
629,631c627
< wget --quiet https://gpg.hestiacp.com/deb_signing.key -O /tmp/deb_signing.key
< APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key add /tmp/deb_signing.key > /dev/null 2>&1
< rm /tmp/deb_signing.key
---
> apt-key adv --fetch-keys 'https://gpg.hestiacp.com/deb_signing.key' > /dev/null 2>&1
637,639c633
<     wget --quiet https://www.postgresql.org/media/keys/ACCC4CF8.asc -O /tmp/psql_signing.key
<     APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1 apt-key add /tmp/psql_signing.key > /dev/null 2>&1
<     rm /tmp/psql_signing.key
---
>     apt-key adv --fetch-keys 'https://www.postgresql.org/media/keys/ACCC4CF8.asc' > /dev/null 2>&1
```

**Comment by ScIT-Raphael** [comment]
Hi @kpapad904 
Once again, thanks for your awesome input! I'll check your requested changes with our dev team and due a propper implementation. I'll gibe you a feedback as soon as the work is done.

**Comment by FalzoMAD** [comment]
+1
